Makefile

```diff
@@ -247,89 +236,45 @@ apply-dev-config:
 	./scripts/create-deployment-blueprint.sh \
-	  --destination "$(DEPLOYCONFIGDIR)" \
+	  --destination "$(DEPLOYPORCHCONFIGDIR)" \
 	  --server-image "$(IMAGE_REPO)/$(PORCH_SERVER_IMAGE):$(IMAGE_TAG)" \
 	  --controllers-image "$(IMAGE_REPO)/$(PORCH_CONTROLLERS_IMAGE):$(IMAGE_TAG)" \
 	  --function-image "$(IMAGE_REPO)/$(PORCH_FUNCTION_RUNNER_IMAGE):$(IMAGE_TAG)" \
 	  --wrapper-server-image "$(IMAGE_REPO)/$(PORCH_WRAPPER_SERVER_IMAGE):$(IMAGE_TAG)" \
-	  --enabled-reconcilers "$(ENABLED_RECONCILERS)" \
-	  --project "$(GCP_PROJECT_ID)"
+	  --enabled-reconcilers "$(ENABLED_RECONCILERS)"
 
 .PHONY: deploy
 deploy: deployment-config
-	kubectl apply -R -f $(DEPLOYCONFIGDIR)
+	kubectl apply -R -f $(DEPLOYPORCHCONFIGDIR)
 
 .PHONY: push-and-deploy
 push-and-deploy: push-images deploy
 
-# Builds deployment config without configuring GCP workload identity for
-# Porch server. This is sufficient for working with GitHub repositories.
-# Workload identity is currently required for Porch to integrate with GCP
-# Container and Artifact Registries; for those use cases, use the make
-# targets without the `-no-sa` suffix (i.e. `deployment-config`,
-# `push-and-deploy` etc.)
-.PHONY: deployment-config-no-sa
-deployment-config-no-sa:
-	mkdir -p $(DEPLOYCONFIG_NO_SA_DIR)
-	find $(DEPLOYCONFIG_NO_SA_DIR) ! -name 'resourcegroup.yaml' -type f -exec rm -f {} +
-	./scripts/create-deployment-blueprint.sh \
-	  --destination "$(DEPLOYCONFIG_NO_SA_DIR)" \
-	  --server-image "$(IMAGE_REPO)/$(PORCH_SERVER_IMAGE):$(IMAGE_TAG)" \
-	  --controllers-image "$(IMAGE_REPO)/$(PORCH_CONTROLLERS_IMAGE):$(IMAGE_TAG)" \
-	  --function-image "$(IMAGE_REPO)/$(PORCH_FUNCTION_RUNNER_IMAGE):$(IMAGE_TAG)" \
-	  --wrapper-server-image "$(IMAGE_REPO)/$(PORCH_WRAPPER_SERVER_IMAGE):$(IMAGE_TAG)" \
-	  --enabled-reconcilers "$(ENABLED_RECONCILERS)"
-
-.PHONY: deploy-no-sa
-deploy-no-sa: deployment-config-no-sa
-	kubectl apply -R -f $(DEPLOYCONFIG_NO_SA_DIR)
-
-.PHONY: push-and-deploy-no-sa
-push-and-deploy-no-sa: push-images deploy-no-sa
-
 KIND_CONTEXT_NAME ?= kind
 
 .PHONY: run-in-kind
 run-in-kind: IMAGE_REPO=porch-kind
 run-in-kind:
-	make build-images
+  ifeq ($(SKIP_IMG_BUILD), false)
+	make build-images; 
+  endif
 	kind load docker-image $(IMAGE_REPO)/$(PORCH_SERVER_IMAGE):${IMAGE_TAG} -n ${KIND_CONTEXT_NAME}
 	kind load docker-image $(IMAGE_REPO)/$(PORCH_CONTROLLERS_IMAGE):${IMAGE_TAG} -n ${KIND_CONTEXT_NAME}
 	kind load docker-image $(IMAGE_REPO)/$(PORCH_FUNCTION_RUNNER_IMAGE):${IMAGE_TAG} -n ${KIND_CONTEXT_NAME}
 	kind load docker-image $(IMAGE_REPO)/$(PORCH_WRAPPER_SERVER_IMAGE):${IMAGE_TAG} -n ${KIND_CONTEXT_NAME}
 	kind load docker-image $(IMAGE_REPO)/$(TEST_GIT_SERVER_IMAGE):${IMAGE_TAG} -n ${KIND_CONTEXT_NAME}
 	make deployment-config
-	KUBECONFIG=$(KUBECONFIG) kubectl apply --wait --recursive --filename $(DEPLOYCONFIGDIR)
+	KUBECONFIG=$(KUBECONFIG) kubectl apply --wait --recursive --filename $(DEPLOYPORCHCONFIGDIR)
```

Comment:
I suggest to replace `kubectl apply` in the make target `run-in-kind`, at line 273 with the following:
```
kpt fn render $(DEPLOYCONFIGDIR)
kpt live init $(DEPLOYCONFIGDIR) || true
kpt live apply --inventory-policy=adopt $(DEPLOYCONFIGDIR)
```
here is the reason why:
- `make deployment-config` will prepare a kpt package in `.build/deploy`, 
- it will also preserve the list of K8s resources that were already applied to the (kind) cluster via `kpt live apply` (indirectly, by keeping the `resourcegroup.yaml` file)
- let's assume that we indeed use `kpt live apply` instead of `kubectl apply` to deploy the kpt package
- if we do modifications in the `.build/deploy` kpt package, then `kpt live apply` is able to calculate the difference between the modified kpt package and the deployed resources much intelligently than `kubectl apply`. It will actually delete resources from the cluster if no longer needed, etc.
- this allows us to introduce new make targets in the future that modify the `.build/deploy` kpt package (i.e. remove porch-server from the package like setup.sh does, or remove the `porch-controllers` deployment, or put back everything into the deploy package, etc.) and then apply it to the cluster seamlessly.

Full disclosure, this exactly my dev process right now, so the proposal might be a bit biased :)

Reply:
Done